OrbitCore/LinuxPerfUtils.h

```diff
@@ -9,51 +9,118 @@
+#include <unistd.h>
 
 //-----------------------------------------------------------------------------
 namespace LinuxPerfUtils
 {
-    // TODO: these constants should be made available in some settings
-    static constexpr uint32_t STACK_SIZE = (1u << 13u) - 8;
-    // Sample stack- and instruction pointer (used later for stack unwinding)
-    static constexpr uint64_t SAMPLE_REGS_USER = (0x1 << PERF_REG_X86_SP) | (0x1 << PERF_REG_X86_IP);
-
-    // This must be in sync with the perf_event_record and the perf_sample_id stucts.
-    static constexpr uint64_t SAMPLE_TYPE_FLAGS = 
-        PERF_SAMPLE_TID | 
-        PERF_SAMPLE_TIME | 
-        PERF_SAMPLE_CPU | 
-        0u;
+    // This must be in sync with the perf_event_record and the perf_sample_id structs.
+    static constexpr uint64_t SAMPLE_TYPE_FLAGS =
+        PERF_SAMPLE_TID |
+        PERF_SAMPLE_TIME |
+        PERF_SAMPLE_CPU;
+
+    // TODO: These constants should be made available in some settings.
+    // Max to pass to perf_event_open without getting an error is (1u << 16u) - 8,
+    // because the kernel stores this in a short and because of alignment reasons.
+    // But the size the kernel actually returns is smaller, because the maximum
+    // size of the entire record the kernel is willing to return is (1u << 16u) - 8.
+    // If we want the size we pass to concide with the size we get, we need to
+    // pass a lower value. For the current layout of perf_record_with_stack,
+    // the maximum size is 65312, but let's leave some extra room.
+    static constexpr uint16_t SAMPLE_STACK_USER_SIZE = 65000;
+
+    // Sample stack pointer and instruction pointer only.
+    static constexpr uint64_t SAMPLE_REGS_USER_SP_IP =
+        (1lu << PERF_REG_X86_SP) |
+        (1lu << PERF_REG_X86_IP);
+
+    // Sample all pointers: they might all be necessary later for DWARF-based stack unwinding.
+    static constexpr uint64_t SAMPLE_REGS_USER_ALL =
+        (1lu << PERF_REG_X86_AX) |
+        (1lu << PERF_REG_X86_BX) |
+        (1lu << PERF_REG_X86_CX) |
+        (1lu << PERF_REG_X86_DX) |
+        (1lu << PERF_REG_X86_SI) |
+        (1lu << PERF_REG_X86_DI) |
+        (1lu << PERF_REG_X86_BP) |
+        (1lu << PERF_REG_X86_SP) |
+        (1lu << PERF_REG_X86_IP) |
+        (1lu << PERF_REG_X86_FLAGS) |
+        (1lu << PERF_REG_X86_CS) |
+        (1lu << PERF_REG_X86_SS) |
+        (1lu << PERF_REG_X86_R8) |
+        (1lu << PERF_REG_X86_R9) |
+        (1lu << PERF_REG_X86_R10) |
+        (1lu << PERF_REG_X86_R11) |
+        (1lu << PERF_REG_X86_R12) |
+        (1lu << PERF_REG_X86_R13) |
+        (1lu << PERF_REG_X86_R14) |
+        (1lu << PERF_REG_X86_R15);
 
     // This struct must be in sync with the SAMPLE_TYPE_FLAGS.
     struct perf_sample_id {
-        uint32_t pid, tid;  /* if PERF_SAMPLE_TID set */
-        uint64_t time;      /* if PERF_SAMPLE_TIME set */
-        uint32_t cpu, res;  /* if PERF_SAMPLE_CPU set */
+        uint32_t pid, tid;  /* if PERF_SAMPLE_TID */
```

Comment:
The /*if ... */ comment needs some explanatory comment.

Reply:
Added some minimal explanation.